Execution of kumuluzee-maven-plugin:repackage fails with Maven 3.8.2

Execution of kumuluzee-maven-plugin:repackage fails with Maven 3.8.2 (latest release) while building sample projects from [kumuluz/kumuluzee-samples](/kumuluz/kumuluzee-samples)

The execution succeeded when Maven 3.8.1 (or earlier version of Maven) was used

Complete execution log:
```
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /home/ikysil/.sdkman/candidates/maven/3.8.2
Java version: 11.0.12, vendor: Amazon.com Inc., runtime: /home/ikysil/.sdkman/candidates/java/11.0.12.7.1-amzn
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-86-generic", arch: "amd64", family: "unix"
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------< com.kumuluz.ee.samples:jax-rs >--------------------
[INFO] Building KumuluzEE JAX-RS sample 3.13.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ jax-rs ---
[INFO] Deleting /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ jax-rs ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ jax-rs ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ jax-rs ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ jax-rs ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ jax-rs ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ jax-rs ---
[INFO] Building jar: /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/jax-rs-3.13.0-SNAPSHOT.jar
[INFO] 
[INFO] --- kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage (package) @ jax-rs ---
[INFO] Checking if project meets the preconditions.
[INFO] Copying kumuluzee-core-3.13.0-SNAPSHOT.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/kumuluzee-core-3.13.0-SNAPSHOT.jar
[INFO] Copying kumuluzee-common-3.13.0-SNAPSHOT.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/kumuluzee-common-3.13.0-SNAPSHOT.jar
[INFO] Copying jakarta.annotation-api-1.3.5.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.annotation-api-1.3.5.jar
[INFO] Copying jakarta.servlet-api-4.0.4.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.servlet-api-4.0.4.jar
[INFO] Copying jakarta.transaction-api-1.3.3.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.transaction-api-1.3.3.jar
[INFO] Copying agroal-api-1.9.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/agroal-api-1.9.jar
[INFO] Copying agroal-hikari-1.9.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/agroal-hikari-1.9.jar
[INFO] Copying HikariCP-3.4.5.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/HikariCP-3.4.5.jar
[INFO] Copying agroal-pool-1.9.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/agroal-pool-1.9.jar
[INFO] Copying snakeyaml-1.27.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/snakeyaml-1.27.jar
[INFO] Copying kumuluzee-servlet-jetty-3.13.0-SNAPSHOT.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/kumuluzee-servlet-jetty-3.13.0-SNAPSHOT.jar
[INFO] Copying jetty-alpn-java-server-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-alpn-java-server-10.0.6.jar
[INFO] Copying jetty-io-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-io-10.0.6.jar
[INFO] Copying slf4j-api-2.0.0-alpha1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/slf4j-api-2.0.0-alpha1.jar
[INFO] Copying jetty-alpn-java-client-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-alpn-java-client-10.0.6.jar
[INFO] Copying jetty-alpn-client-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-alpn-client-10.0.6.jar
[INFO] Copying jetty-webapp-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-webapp-10.0.6.jar
[INFO] Copying jetty-servlet-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-servlet-10.0.6.jar
[INFO] Copying jetty-security-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-security-10.0.6.jar
[INFO] Copying jetty-xml-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-xml-10.0.6.jar
[INFO] Copying jetty-annotations-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-annotations-10.0.6.jar
[INFO] Copying jetty-plus-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-plus-10.0.6.jar
[INFO] Copying asm-9.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/asm-9.1.jar
[INFO] Copying asm-commons-9.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/asm-commons-9.1.jar
[INFO] Copying asm-tree-9.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/asm-tree-9.1.jar
[INFO] Copying asm-analysis-9.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/asm-analysis-9.1.jar
[INFO] Copying jetty-jndi-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-jndi-10.0.6.jar
[INFO] Copying jetty-util-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-util-10.0.6.jar
[INFO] Copying http2-server-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/http2-server-10.0.6.jar
[INFO] Copying http2-common-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/http2-common-10.0.6.jar
[INFO] Copying http2-hpack-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/http2-hpack-10.0.6.jar
[INFO] Copying jetty-server-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-server-10.0.6.jar
[INFO] Copying jetty-servlet-api-4.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-servlet-api-4.0.6.jar
[INFO] Copying jetty-http-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-http-10.0.6.jar
[INFO] Copying jetty-alpn-server-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-alpn-server-10.0.6.jar
[INFO] Copying http2-http-client-transport-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/http2-http-client-transport-10.0.6.jar
[INFO] Copying jetty-client-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jetty-client-10.0.6.jar
[INFO] Copying http2-client-10.0.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/http2-client-10.0.6.jar
[INFO] Copying slf4j-jdk14-2.0.0-alpha1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/slf4j-jdk14-2.0.0-alpha1.jar
[INFO] Copying kumuluzee-jax-rs-jersey-3.13.0-SNAPSHOT.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/kumuluzee-jax-rs-jersey-3.13.0-SNAPSHOT.jar
[INFO] Copying jersey-container-servlet-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-container-servlet-2.34.jar
[INFO] Copying jersey-container-servlet-core-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-container-servlet-core-2.34.jar
[INFO] Copying jersey-common-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-common-2.34.jar
[INFO] Copying osgi-resource-locator-1.0.3.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/osgi-resource-locator-1.0.3.jar
[INFO] Copying jersey-server-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-server-2.34.jar
[INFO] Copying jersey-client-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-client-2.34.jar
[INFO] Copying jakarta.validation-api-2.0.2.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.validation-api-2.0.2.jar
[INFO] Copying jakarta.ws.rs-api-2.1.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.ws.rs-api-2.1.6.jar
[INFO] Copying jersey-media-sse-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-media-sse-2.34.jar
[INFO] Copying jakarta.inject-2.6.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.inject-2.6.1.jar
[INFO] Copying jersey-media-json-jackson-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-media-json-jackson-2.34.jar
[INFO] Copying jersey-entity-filtering-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-entity-filtering-2.34.jar
[INFO] Copying jackson-annotations-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-annotations-2.12.0.jar
[INFO] Copying jackson-databind-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-databind-2.12.0.jar
[INFO] Copying jackson-core-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-core-2.12.0.jar
[INFO] Copying jackson-module-jaxb-annotations-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-module-jaxb-annotations-2.12.0.jar
[INFO] Copying jakarta.xml.bind-api-2.3.2.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.xml.bind-api-2.3.2.jar
[INFO] Copying jakarta.activation-api-1.2.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.activation-api-1.2.1.jar
[INFO] Copying jersey-hk2-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-hk2-2.34.jar
[INFO] Copying hk2-locator-2.6.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/hk2-locator-2.6.1.jar
[INFO] Copying aopalliance-repackaged-2.6.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/aopalliance-repackaged-2.6.1.jar
[INFO] Copying hk2-api-2.6.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/hk2-api-2.6.1.jar
[INFO] Copying hk2-utils-2.6.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/hk2-utils-2.6.1.jar
[INFO] Copying jersey-cdi1x-servlet-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-cdi1x-servlet-2.34.jar
[INFO] Copying jersey-cdi1x-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-cdi1x-2.34.jar
[INFO] Copying jakarta.enterprise.cdi-api-2.0.2.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.enterprise.cdi-api-2.0.2.jar
[INFO] Copying jakarta.el-api-3.0.3.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.el-api-3.0.3.jar
[INFO] Copying jakarta.interceptor-api-1.2.5.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.interceptor-api-1.2.5.jar
[INFO] Copying jakarta.ejb-api-3.2.6.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.ejb-api-3.2.6.jar
[INFO] Copying jakarta.inject-api-1.0.3.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jakarta.inject-api-1.0.3.jar
[INFO] Copying javassist-3.27.0-GA.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/javassist-3.27.0-GA.jar
[INFO] Copying jackson-jaxrs-json-provider-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-jaxrs-json-provider-2.12.0.jar
[INFO] Copying jackson-jaxrs-base-2.12.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jackson-jaxrs-base-2.12.0.jar
[INFO] Copying jersey-jetty-connector-2.34.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jersey-jetty-connector-2.34.jar
[INFO] Copying activation-1.1.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/activation-1.1.1.jar
[INFO] Copying jaxb-api-2.3.1.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/jaxb-api-2.3.1.jar
[INFO] Copying javax.activation-api-1.2.0.jar to /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/classes/lib/javax.activation-api-1.2.0.jar
[INFO] Unpacking kumuluzee-loader dependency.
[INFO] Building jar: /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/jax-rs-3.13.0-SNAPSHOT-uber.jar
[INFO] Repackaging jar: /home/ikysil/workshop/kumuluz/kumuluzee-samples/jax-rs/target/jax-rs-3.13.0-SNAPSHOT.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.520 s
[INFO] Finished at: 2021-09-25T22:15:53+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.kumuluz.ee:kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage (package) on project jax-rs: Execution package of goal com.kumuluz.ee:kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage failed.: UnsupportedOperationException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.kumuluz.ee:kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage (package) on project jax-rs: Execution package of goal com.kumuluz.ee:kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution package of goal com.kumuluz.ee:kumuluzee-maven-plugin:3.13.0-SNAPSHOT:repackage failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.UnsupportedOperationException
    at java.util.Collections$UnmodifiableCollection.remove (Collections.java:1063)
    at com.kumuluz.ee.maven.plugin.AbstractPackageMojo.renameJars (AbstractPackageMojo.java:207)
    at com.kumuluz.ee.maven.plugin.AbstractPackageMojo.repackage (AbstractPackageMojo.java:66)
    at com.kumuluz.ee.maven.plugin.RepackageMojo.execute (RepackageMojo.java:47)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```

--- a/pom.xml
+++ b/pom.xml
@@ -63,9 +63,9 @@
         <agroal.version>1.9</agroal.version>
         <yaml.version>1.27</yaml.version>
         <junit.version>4.13.1</junit.version>
-        <maven.plugin.api.version>3.6.3</maven.plugin.api.version>
-        <maven-plugin-annotations.version>3.6.0</maven-plugin-annotations.version>
-        <mojo-executor.version>2.3.1</mojo-executor.version>
+        <maven.plugin.api.version>3.8.2</maven.plugin.api.version>
+        <maven-plugin-annotations.version>3.6.1</maven-plugin-annotations.version>
+        <mojo-executor.version>2.3.2</mojo-executor.version>
 
         <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
         <maven-resources-plugin.version>3.2.0</maven-resources-plugin.version>

--- a/tools/maven-plugin/src/main/java/com/kumuluz/ee/maven/plugin/AbstractPackageMojo.java
+++ b/tools/maven-plugin/src/main/java/com/kumuluz/ee/maven/plugin/AbstractPackageMojo.java
@@ -20,9 +20,10 @@
  */
 package com.kumuluz.ee.maven.plugin;
 
-import org.apache.maven.artifact.Artifact;
+import org.apache.maven.execution.MavenSession;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
 
 import java.io.IOException;
 import java.io.InputStream;
@@ -32,7 +33,6 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.*;
 import java.security.CodeSource;
 import java.security.ProtectionDomain;
-import java.util.List;
 import java.util.jar.JarFile;
 
 import static org.twdata.maven.mojoexecutor.MojoExecutor.*;
@@ -155,6 +155,14 @@ public abstract class AbstractPackageMojo extends AbstractCopyDependenciesMojo {
     }
 
     private void packageJar() throws MojoExecutionException {
+
+        // Execute maven-jar-plugin in a separate execution environment, in order not to pollute the current project with
+        // attached artifact. Attached artifact will replace the current main artifact after rename, but reference in
+        // project.getAttachedArtifacts() cannot be removed after it's been added by maven-jar-plugin.
+        MavenProject projectClone = new MavenProject(project);
+        MavenSession sessionClone = session.clone();
+        sessionClone.setCurrentProject(projectClone);
+
         executeMojo(
                 plugin(
                         groupId("org.apache.maven.plugins"),
@@ -173,7 +181,7 @@ public abstract class AbstractPackageMojo extends AbstractCopyDependenciesMojo {
                                 )
                         )
                 ),
-                executionEnvironment(project, session, buildPluginManager)
+                executionEnvironment(projectClone, sessionClone, buildPluginManager)
         );
     }
 
@@ -199,14 +207,6 @@ public abstract class AbstractPackageMojo extends AbstractCopyDependenciesMojo {
                         sourcePath2.resolveSibling(finalName + ".jar"),
                         StandardCopyOption.REPLACE_EXISTING
                 );
-            }
-
-            List<Artifact> artifacts = project.getAttachedArtifacts();
-            for (Artifact artifact : project.getAttachedArtifacts()) {
-                if (artifact.hasClassifier() && artifact.getClassifier().equals("uber")) {
-                    artifacts.remove(artifact);
-                    break;
-                }
             }
         } catch (IOException e) {
             throw new MojoExecutionException("Unable to rename the final build artifact.");

